### PR TITLE
fix / doubleSpacing

### LINKED ISSUE #197: Double spacing typo throw unexpected error
- [ ] Crashes when the following task in entered: `from 23__june`, where `_` is a space character
- [ ] Unhanded exception when for: `from 23__jun to 24 jun go party`, where `_` is a space character

### PR DESCRIPTION
**Pull request by bsmmoon**
closes #197

### CODE REVIEW

File: DoLah/Parser/ParserLibrary.cpp [hunk 1]
```diff
@@ -106,10 +106,28 @@ namespace DoLah {
...
+            str.replace(f, substr.length(), "");
+            f = str.find(substr);
+        }
+        return str;
+    }
+
+    std::string ParserLibrary::removeDoubleSpacing(std::string str) {
+        std::string substr = "  ";
+        return stringRemoveAll(str, substr);
+    }
```
Comment (9gix) [inline DoLah/Parser/ParserLibrary.cpp]: remove double spacing? only? triple spacing? etc...?

File: DoLah/Parser/CommandParser.cpp [hunk 2]
```diff
@@ -155,6 +155,7 @@ namespace DoLah {
 
 
     AbstractCommand* CommandParser::parse(std::string input) {
+        input = ParserLibrary::removeDoubleSpacing(input);
```
Comment (9gix) [inline DoLah/Parser/CommandParser.cpp]: how if the user want to input a task with double spacing?

Comment (bsmmoon) [inline DoLah/Parser/CommandParser.cpp]: will simply ignore it.

Quite a lot of software behave in similar manner I believe.

File: DoLah/Parser/ParserLibrary.cpp (hunk 1 above)
Comment (bsmmoon) [inline DoLah/Parser/ParserLibrary.cpp]: It removed everything above 2.

**Pull request merged**